BraveWallet/Crypto/Stores/SendTokenStore.swift

```diff
@@ -185,7 +191,15 @@ public class SendTokenStore: ObservableObject {
       || (allTokens.first(where: { $0.contractAddress.lowercased() == normalizedSendAddress }) != nil) {
       addressError = .contractAddress
     } else {
-      addressError = nil
+      keyringService.checksumEthAddress(sendAddress) { [self] checksumAddress in
+        if sendAddress == checksumAddress {
+          addressError = nil
+        } else if sendAddress.removingHexPrefix.lowercased() == sendAddress.removingHexPrefix || sendAddress.removingHexPrefix.uppercased() == sendAddress.removingHexPrefix {
```

Comment:
Just do `caseInsentiveCompare` instead of doing both uppercase/lowercase

Reply:
based on the requirement , this is checking if the address is all in lowercase or all in uppercase. if the address is mixed with lowercase and uppercase then either it has to be the same as the checksum address (the first condition) or it has invalid checksum address (the last condition)